test/libsolidity/SolidityABIJSON.cpp

```diff
@@ -292,6 +308,8 @@ BOOST_AUTO_TEST_CASE(events)
 	{
 		"name": "f",
 		"constant": false,
+		"view": false,
```

Comment:
From yesterday's call:

We have the following types (from most restricted / smallest subset of instructions): pure, view, non-payable, regular/payable.

It can be represented as `state-access` with levels of `none`, `read-only`, `non-payable`, `full`.

It may be beneficial to make separation of state and memory access. Think about `internal pure` functions which work on a memory reference - can they change data?